File: app/src/main/java/com/xuie/android/ui/palette/PaletteFragment.java
package com.xuie.android.ui.palette;


import android.os.Bundle;
import android.util.TypedValue;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.xuie.android.R;

import java.util.Objects;

/**
 * A simple {@link Fragment} subclass.
 *
 * @author xuie
 */
public class PaletteFragment extends Fragment {
    private static final String ARG_POSITION = "position";

    private int position;
    private static int[] drawables = {R.mipmap.one, R.mipmap.two, R.mipmap.four, R.mipmap.three, R.mipmap.five};

    public static PaletteFragment newInstance(int position) {
        PaletteFragment f = new PaletteFragment();
        Bundle b = new Bundle();
        b.putInt(ARG_POSITION, position);
        f.setArguments(b);
        return f;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        position = getArguments().getInt(ARG_POSITION);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle
            savedInstanceState) {

        FrameLayout.LayoutParams params = new FrameLayout.LayoutParams(FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams
                .MATCH_PARENT);

        FrameLayout fl = new FrameLayout(Objects.requireNonNull(getActivity()));
        fl.setLayoutParams(params);
        fl.setBackgroundResource(drawables[position]);
        final int margin = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 8,
                getResources().getDisplayMetrics());

        TextView v = new TextView(getActivity());
        params.setMargins(margin, margin, margin, margin);
        v.setLayoutParams(params);
        v.setLayoutParams(params);
        v.setGravity(Gravity.BOTTOM);
        v.setText("CARD " + (position + 1));

        fl.addView(v);
        return fl;
    }

    public static int getBackgroundBitmapPosition(int selectViewPagerItem) {
        return drawables[selectViewPagerItem];
    }

}

File: app/src/main/java/com/xuie/android/ui/palette/PaletteFragment.kt
package com.xuie.android.ui.palette


import android.os.Bundle
import android.util.TypedValue
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import com.xuie.android.R
import java.util.*

/**
 * A simple [Fragment] subclass.
 *
 * @author xuie
 */
class PaletteFragment : Fragment() {

  private var position: Int = 0

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    position = arguments!!.getInt(ARG_POSITION)
  }

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

    val params = FrameLayout.LayoutParams(FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams
        .MATCH_PARENT)

    val fl = FrameLayout(Objects.requireNonNull<FragmentActivity>(activity))
    fl.layoutParams = params
    fl.setBackgroundResource(drawables[position])
    val margin = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 8f,
        resources.displayMetrics).toInt()

    val v = TextView(activity)
    params.setMargins(margin, margin, margin, margin)
    v.layoutParams = params
    v.layoutParams = params
    v.gravity = Gravity.BOTTOM
    v.text = "CARD " + (position + 1)

    fl.addView(v)
    return fl
  }

  companion object {
    private const val ARG_POSITION = "position"
    private val drawables = intArrayOf(R.mipmap.one, R.mipmap.two, R.mipmap.four, R.mipmap.three, R.mipmap.five)

    fun newInstance(position: Int): PaletteFragment {
      val f = PaletteFragment()
      val b = Bundle()
      b.putInt(ARG_POSITION, position)
      f.arguments = b
      return f
    }

    fun getBackgroundBitmapPosition(selectViewPagerItem: Int): Int {
      return drawables[selectViewPagerItem]
    }
  }

}
